src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -273,18 +273,161 @@ public void Shuffle<T>(T[] values)
         /// </remarks>
         public void Shuffle<T>(Span<T> values)
         {
-            int n = values.Length;
+            var chooser = new IncreasingUniform(this);
+            for (int i = 1; i < values.Length; i++)
+            {
+                int index = chooser.NextIndex();
+
+                // This pattern is faster than tuple deconstruction for large structs
```

Comment:
is that something the compiler could/should improve?

Reply:
It was mentioned [here](https://github.com/dotnet/runtime/pull/70944#discussion_r962206857), my understand is that it will be fixed after C# 11